File: 122.Best Time to Buy and Sell Stock II/122.py
class Solution:
    def maxProfit(self, prices) -> int:
        if len(prices) == 0:
            return 0
        profit = 0
        start = prices[0]
        for i in prices:
            if i>start:
                profit += i-start
            start = i

        return profit

a = Solution()
print(a.maxProfit([7,6,4,3,1]))

############################################
# return sum(max(prices[i + 1] - prices[i], 0) for i in range(len(prices) - 1))
# return sum([b-a for a,b in zip(prices,prices[1:]) if b-a > 0])

# 这两个一行的语句都很pythonic

File: 122.Best Time to Buy and Sell Stock II/122.go
package main

import "fmt"

func main(){
	fmt.Println(maxProfit([]int{7,6,4,3,1}))
}

func maxProfit(prices []int) int {
    if len(prices) < 2{
		return 0
	}
	start := prices[0]
	profit := 0
	for _,value := range prices{
		if value>start{
			profit += (value-start)
		}
		start = value
	}
	return profit
}

func maxProfit2(prices []int) int {

	var ret int
	
	for i := 1; i < len(prices); i++ {
		if(prices[i] > prices[i-1]) {
			ret += prices[i] - prices[i-1]
		}
	}
	
	return ret
	}